middleware/git/git.go

```diff
@@ -0,0 +1,156 @@
+package git
+
+import (
+	"fmt"
+	"github.com/mholt/caddy/middleware"
+	"net/http"
+	"net/url"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Git represents a middleware instance that pulls git repository.
+type Git struct {
+	Next middleware.Handler
+	Repo *Repo
+}
+
+// ServeHTTP satisfies the middleware.Handler interface.
+func (g Git) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+	if err := g.Repo.Pull(); err != nil {
+		return 500, err
+	}
+	return g.Next.ServeHTTP(w, r)
+}
+
+// New creates a new instance of git middleware.
+func New(c middleware.Controller) (middleware.Middleware, error) {
+	repo, err := parse(c)
+	if err != nil {
+		return nil, err
+	}
+	err = repo.Pull()
+	return func(next middleware.Handler) middleware.Handler {
+		return Git{Next: next, Repo: repo}
+	}, err
+}
+
+func parse(c middleware.Controller) (*Repo, error) {
+	repo := &Repo{Branch: "master", Interval: DefaultInterval, Path: c.Root()}
+
+	for c.Next() {
+		args := c.RemainingArgs()
+
+		switch len(args) {
+		case 2:
+			repo.Path = args[1]
+			fallthrough
+		case 1:
+			repo.Url = args[0]
+		}
+
+		for c.NextBlock() {
+			switch c.Val() {
+			case "repo":
+				if !c.NextArg() {
+					return nil, c.ArgErr()
+				}
+				repo.Url = c.Val()
+			case "path":
+				if !c.NextArg() {
+					return nil, c.ArgErr()
+				}
+				repo.Path = c.Val()
```

Comment:
What do you think about making the path relative to site root, since it specifies where in the site to clone the repo?